frontend/src/components/playground/OutputDocumentSection.tsx

```diff
@@ -2,23 +2,66 @@ import Container from "@cloudscape-design/components/container";
 import Header from "@cloudscape-design/components/header";
 import SpaceBetween from "@cloudscape-design/components/space-between";
 import Box from "@cloudscape-design/components/box";
+import Alert from "@cloudscape-design/components/alert";
+import Spinner from "@cloudscape-design/components/spinner";
 import { usePlayground } from "@/context/PlaygroundContext";
 import { DocumentItemWithPopoverCodeView } from "./DocumentItemWithPopoverCodeView";
+import { ReactNode } from "react";
 
-// Inner component that uses the usePlayground hook
 export default function OutputDocumentSection() {
   const { state } = usePlayground();
+  const { isProcessingTransformations, transformationErrors } = state;
+
+  const renderOutputContent = (): ReactNode => {
+    if (isProcessingTransformations) {
+      return (
+        <Box textAlign="center">
+          <Spinner size="large" />
+          <Box variant="p" color="text-status-info">
+            Processing transformations...
+          </Box>
+        </Box>
+      );
+    }
+
+    if (state.outputDocuments.length === 0) {
+      return <Box>No output documents.</Box>;
+    }
+
+    return state.outputDocuments.map((doc) => (
+      <DocumentItemWithPopoverCodeView key={doc.id} document={doc} />
+    ));
+  };
 
   return (
     <Container header={<Header variant="h3">Output Documents</Header>}>
       <SpaceBetween size="m">
-        {state.outputDocuments.length === 0 ? (
-          <Box>No output documents.</Box>
-        ) : (
-          state.outputDocuments.map((doc) => (
-            <DocumentItemWithPopoverCodeView key={doc.id} document={doc} />
-          ))
+        {transformationErrors.length > 0 && (
```

Comment:
I'm not sure theming is supported on UL/LI can we just put these in their own `<Box>` to make sure they better use horizontal space? 
![image](https://github.com/user-attachments/assets/618568a4-0909-41a6-b12e-6481dbae168c)

Reply:
Done -- this is the new appearance

<img width="359" alt="Screenshot 2025-05-14 at 4 24 57 PM" src="https://github.com/user-attachments/assets/e8598086-6c0e-44fa-8208-f2f15311ff7e" />